validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -220,6 +228,29 @@ private List<Metric> listMetricFromCloudWatch(
     return result;
   }
 
+    private boolean metricEquals(Metric expected, Metric actual) {
+        if (expected.getNamespace().equals(actual.getNamespace())
+                && expected.getMetricName().equals(actual.getMetricName())) {
+            if (expected.getDimensions().size() == actual.getDimensions().size()) {
+                for (int i = 0; i < expected.getDimensions().size(); i++) {
+                    if (!dimensionEquals(expected.getDimensions().get(i), actual.getDimensions().get(i))) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean dimensionEquals(Dimension expected, Dimension actual) {
+        if (expected.getName().equals(actual.getName())) {
+            return ANY_VALUE.equals(expected.getValue()) ||
+                    expected.getValue().equals(actual.getValue());
+        }
+        return false;
+    }
```

Comment:
out of scope nit: Looks like we have an opportunity to implement RegEx instead of relying on only `ANY_VALUE`. If it's not too much effort the validations could implement some regex and this could apply to all metrics instead of having to split the logic based on any value vs. exact match.

Definitely not a requirement, just a thought

Reply:
Supporting regex means we can no longer utilize hash set to remove the exact match metrics quickly. It may bring performance concern because the algorithm complexity will increase.